pilot/pkg/proxy/envoy/v2/discovery.go

```diff
@@ -311,16 +311,18 @@ func (s *DiscoveryServer) periodicRefreshMetrics() {
 // Push is called to push changes on config updates using ADS. This is set in DiscoveryService.Push,
 // to avoid direct dependencies.
 func (s *DiscoveryServer) Push(full bool, edsUpdates map[string]*EndpointShardsByService) {
-	if !full {
-		adsLog.Infof("XDS Incremental Push EDS:%d", len(edsUpdates))
-		go s.AdsPushAll(versionInfo(), s.globalPushContext(), false, edsUpdates)
-		return
-	}
 	// Reset the status during the push.
 	pc := s.globalPushContext()
 	if pc != nil {
 		pc.OnConfigChange()
 	}
+
+	if !full {
```

Comment:
I don't think EDS change should cause onConfigChange() - just needs to update metrics related to clusters.

Reply:
This is `OnConfigChange` does now, set pushStatus and update metrics.
```
// OnConfigChange is called when a config change is detected.
func (ps *PushContext) OnConfigChange() {
	LastPushMutex.Lock()
	LastPushStatus = ps
	LastPushMutex.Unlock()
	ps.UpdateMetrics()
}
```

Reply:
Only by this we can get the accurate status info from debug endpoint.